impeller/compiler/shader_lib/impeller/gradient.glsl

```diff
@@ -7,6 +7,29 @@
 
 #include <impeller/texture.glsl>
 
+/// Compute the t value for a conical gradient at point `p` between the 2
+/// circles defined by (c0, r0) and (c1, r1).
+///
+/// This assumes that c0 != c1.
+float IPComputeConicalT(vec2 c0, float r0, vec2 c1, float r1, vec2 p) {
+  float w = 1.0;
+  float result = 0.0;
+  vec2 ab = c1 - c0;
+  float dr = r1 - r0;
+  // Set sample rate to a minimum for the case where c0 and c1 are close.
```

Comment:
Can we handle this with a kEhCloseEnough check in the conical contents? if C0 close to C1, then its just a radial gradient right?

Reply:
in order for it to be the same it has to be c0==c1 and r1 == 0.  If c0==c1 we can do the math easily without the loop but it didn't seem worth the extra complexity.

Reply:
ahh got it.